core/modules/comment/comment.install

```diff
@@ -470,6 +470,35 @@ function comment_update_1007() {
   }
 }
 
+/**
+ * New setting for comment titles. Replacing comment_subject_field
+ * with comment_title_options.
+ */
+function comment_update_1008() {
+  foreach (config_get_names_with_prefix('node.type.') as $config_name) {
+    $config = config($config_name);
+    $comment_title = $config->get('settings.comment_subject_field');
+    if ($comment_title == COMMENT_TITLE_AUTO) {
+      // Set for auto generated titles.
+      $config->set('settings.comment_title_options', COMMENT_TITLE_AUTO);
+    } 
+    else {
+      // Set for custom titles.
+      $config->set('settings.comment_title_options', COMMENT_TITLE_CUSTOM);
+    }
+    $config->clear('settings.comment_subject_field');
+    $config->save();
+  
```

Comment:
There's a syntax error, above "foreach" isn't closed, a `}` is missing here.

Reply:
Dang this coding stuff can be difficult. Fixing now.